TileStache/Mapnik.py

```diff
@@ -74,19 +74,24 @@ class ImageProvider:
         - http://trac.mapnik.org/wiki/XMLConfigReference
     """
 
-    def __init__(self, layer, mapfile, fonts=None, scale_factor=None):
-        """ Initialize Mapnik provider with layer and mapfile.
+    def __init__(self, layer, mapconfig, fonts=None, scale_factor=None):
+        """ Initialize Mapnik provider with layer and mapconfig.
 
-            XML mapfile keyword arg comes from TileStache config,
+            XML mapconfig keyword arg comes from TileStache config,
             and is an absolute path by the time it gets here.
         """
-        maphref = urljoin(layer.config.dirpath, mapfile)
+        maphref = urljoin(layer.config.dirpath, mapconfig)
         scheme, h, path, q, p, f = urlparse(maphref)
 
+        self.mapconfig_is_string = False
```

Comment:
This variable name is not quite clear, because a path is an string too. Also, from what I understand, the need is to avoid using extra files to configure server, so Mapnik objects should be favored instead of XML strings.